### PR TITLE
design: Add correlation design

### PR DESCRIPTION
**Pull request by brancz**
This is a start for designs for correlating signals, starting with metrics to logs and metrics to traces.

@ant31 @bwplotka @s-urbaniak @metalmatze @squat @kakkoyun @LiliC

### CODE REVIEW

File: design/correlation.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+sum(rate(http_requests_total{code="5.."}[5m])) / sum(rate(http_requests_total[5m])) > X
+
+This query gives us a fleet wide aggregation of percentage of 5xx errors. We want to see logs from a container that produced such errors.
+
+Problem: The aggregation lost the required metadata identifying the workload.
+
+topk(10, sum by(cluster, namespace, pod, container) (rate(http_requests_total{code="5.."}[5m])) / sum by(cluster, namespace, pod, container) (rate(http_requests_total[5m])))
+
+With the result of the above query we have enough metadata to jump to logs within the violated -10m - time(alert fired). This significantly narrows down the amount of data that needs to be browsed to continue troubleshooting.
```
Comment (s-urbaniak) [inline design/correlation.md]: With metadata what do we mean concretely? I believe it is simply well-known labels?

Comment (bwplotka) [reply design/correlation.md]: Metadata means here any labels that we can use to query other's signal backend (: 

So looks like `by(cluster, namespace, pod, container) `

**Comment by brancz** [comment]
Yep, I just want to get things going.

File: design/correlation.md (hunk 1 above)
Comment (brancz) [reply design/correlation.md]: Precisely.

**Pull request merged**